test/unit/protocols/jds_line_test.dart

```diff
@@ -0,0 +1,189 @@
+import 'dart:async';
+import 'dart:io';
+
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:hmi_core/hmi_core.dart';
+import 'package:hmi_networking/hmi_networking.dart';
+
+import '../helpers.dart';
+import '../test_data.dart';
+
+
+class FakeLineSocket extends Fake implements LineSocket {
+  final Stream<Uint8List> _fakeStream;
+  final bool _isConnected;
+  final Result<bool> _requestAllResult;
+  final Result<bool> _sendResult;
+
+  FakeLineSocket({
+    Stream<Uint8List> stream = const Stream.empty(),
+    bool isConnected = true, 
+    Result<bool> requestAllResult = const Result(data: true),
+    Result<bool> sendResult = const Result(data: true),
+  }) : _fakeStream = stream,
+    _isConnected = isConnected,
+    _requestAllResult = requestAllResult,
+    _sendResult = sendResult;
+
+  @override
+  Stream<Uint8List> get stream => _fakeStream;
+
+  @override
+  bool get isConnected => _isConnected;
+
+  @override
+  Result<bool> requestAll() => _requestAllResult;
+
+  @override
+  Future<Result<bool>> send(List<int> data) => Future.value(_sendResult);
+
+  @override
+  Future close() => Future.value(true);
+}
+
+void main() {
+  group('JdsLine', () {
+    late FakeLineSocket socket;
+    late JdsLine line;
+
+      test('stream', () async {
+        socket = FakeLineSocket(
+          stream: Stream.fromFuture(
+            Future.delayed(
+              const Duration(milliseconds: 100), 
+              () => encodeDataPoints(testDataPoints),
+            ),
+          ),
+          isConnected: true,
+        );
+        line = JdsLine(lineSocket: socket);
+
+        final receivedEvents = <DsDataPoint>[];
+        line.stream.listen((event) { receivedEvents.add(event); });
+
+        await Future.delayed(const Duration(milliseconds: 100));
+        
+        expect(compareDataPointCollections(receivedEvents, testDataPoints), true);
+      });
+
+      test('requestAll', () async {
```

Comment:
was sent to the server the specific requestAll command  must be verified first of all:
DsCommand(
      dsClass: DsDataClass.requestAll,
      type: DsDataType.bool,
      path: '',
      name: '',
      value: 1,
      status: DsStatus.ok,
      timestamp: DsTimeStamp.now(),
    )

Reply:
Done:
https://github.com/a-givertzman/hmi_networking/pull/3/files#diff-2f289a8438c1fbb1711f3abbd59844e18cb9702b03b49700386d3198685598ae